src/extractor.rs

```diff
@@ -0,0 +1,60 @@
+use std::iter::Iterator;
+use std::iter::Peekable;
+use std::vec::IntoIter;
+
+static FULL_STOP: char = '。';
+
+/// An Iterator extracting _sentences_ from a text node.
+/// We throw away:
+/// - Anything be fore the first [FULL_STOP]
+/// - Anthing where [contains_no_invalid_char] returns `false`
+/// - Anything after the last [FULL_STOP]
+pub struct SentenceExtractor<'a> {
+    sentences: Peekable<IntoIter<&'a str>>,
+}
+
+impl<'a> SentenceExtractor<'a> {
+    pub fn new(sentences: &'a str) -> Self {
+        SentenceExtractor {
+            sentences: extract_sentences(sentences),
+        }
+    }
+}
+
+impl<'a> Iterator for SentenceExtractor<'a> {
+    type Item = &'a str;
+    fn next(&mut self) -> Option<&'a str> {
+        if self.sentences.peek().is_some() {
```

Comment:
why not return `self.sentences.next()` directly? i.e. wouldn't the API of this function be the same without the conditional?